execute.go

```diff
@@ -92,6 +87,34 @@ func executeWithOutputChan(outputChan chan string, env []string, workingDir stri
 	}, nil
 }
 
+// kills a cmd process based on config timeout settings
+func killProcessOnTimeout(cmd *exec.Cmd, conf *effectiveConfig) (int, error) {
+	var exitCode int
+	// 1 deep channel for done
+	done := make(chan error, 1)
+
+	go func() {
+		done <- cmd.Wait()
+	}()
+
+	select {
+	case <-time.After(conf.BuildTimeout()):
+		if err := cmd.Process.Kill(); err != nil {
+			return 0, fmt.Errorf("Failed to kill process: %v", err)
+		}
+		<-done
+	case err := <-done:
+		if err != nil {
+			if exitErr, ok := err.(*exec.ExitError); ok {
```

Comment:
so if there is an error that is not a wait status exit error we just throw it away?  Seems bad.